src/modules/esl-avatar/core/esl-avatar.ts

```diff
@@ -0,0 +1,109 @@
+  @prop('esl:avatar:changed') public AVATAR_CHANGED_EVENT: string;
+
+  /** Source path of the avatar image */
+  @attr() public imageSrc: string;
+  /** The limit number of letters to be displayed in text-only mode */
+  @attr({defaultValue: 2}) public limit: number;
+  /** Policy of loading image that is outside of the viewport */
+  @attr({defaultValue: 'lazy'}) public loading: 'eager' | 'lazy';
+  /** The name of the user for whom the avatar is displayed */
+  @attr({defaultValue: ''}) public username: string;
+
+  /** @readonly Marker of displaying mode with image */
+  @boolAttr({readonly: true}) public withImg: boolean;
+  /** @readonly Marker of displaying mode without image, only text */
+  @boolAttr({readonly: true}) public textOnly: boolean;
+
+  /** @readonly Ready state marker */
+  @boolAttr({readonly: true}) public ready: boolean;
+
+  /** Gets an abbreviation to display in text-only mode and for alt property of image */
+  public get abbr(): string {
+    return this.username.trim()
+      .split(' ')
+      .filter(Boolean)
+      .reduce((acc, el, index) => (index >= this.limit) ? acc : acc + el.slice(0, 1), '');
+  }
+
+  protected override connectedCallback(): void {
+    super.connectedCallback();
+    this.init();
+  }
+
+  protected override attributeChangedCallback(attrName: string, oldVal: string, newVal: string): void {
+    if (!this.connected || oldVal === newVal) return;
+    this.init(true);
+  }
+
+  /** Initializes the avatar */
+  public init(force?: boolean): void {
+    if (this.ready && !force) return;
+    this.buildImageContent();
+    this.initA11y();
+    this.onReady();
+  }
+
+  /** Sets initial a11y attributes */
+  protected initA11y(): void {
+    if (this.$$attr('title') === null) this.title = this.username;
+  }
+
+  /** Builds the image on avatar */
+  protected buildImageContent(): void {
+    const {imageSrc} = this;
+    if (!imageSrc) return this._onImageError();
+
+    const $img = new Image();
+    $img.loading = this.loading || 'lazy';
+    $img.alt = this.abbr;
+    $img.src = imageSrc;
+    $img.onerror = this._onImageError;
+    this.appendContent($img, true);
+  }
+
+  /** Builds the text on avatar */
+  protected buildTextContent(): void {
+    const $text = document.createElement('abbr');
+    $text.textContent = this.abbr;
+    this.appendContent($text, false);
+  }
+
+  /** Appends content to the component */
+  protected appendContent($content: HTMLElement, isImage: boolean): void {
+    this.innerHTML = '';
+    $content.classList.add(`${ESLAvatar.is}-${isImage ? 'img' : 'text'}`);
+    this.appendChild($content);
+    this.$$attr('with-img', isImage);
+    this.$$attr('text-only', !isImage);
+  }
+
+  /** Actions on image loading error */
+  @bind
+  protected _onImageError(): void {
+    this.buildTextContent();
+  }
+
+  /** Actions on complete init and ready component */
+  private onReady(): void {
+    this.$$attr('ready', true);
```

Comment:
Ready = initialized or content (image) is loaded ?

Reply:
reworked